### PR TITLE
Update firefly-helm-charts to FireFly 1.3 and KiND deployment

### PR DESCRIPTION
**Pull request by SamMayWork**
ref: https://github.com/hyperledger/firefly-helm-charts/pull/74

This PR is only a slight change from the core of the PR above with the following differences:

 - Deploys FireFly 1.3
 - Does not directly mirror the Besu helm charts from an external provider, they're now cloned at deploy-time and applied to the cluster

### CODE REVIEW

File: README.md
Comment (EnriqueL8) [inline README.md:37]: Does this work with the new APIs we have introduced in main?

Comment (EnriqueL8) [reply README.md:37]: If not could you please raise an issue as a follow up

File: charts/firefly-evmconnect/Chart.yaml
Comment (EnriqueL8) [inline charts/firefly-evmconnect/Chart.yaml:9]: ???

File: charts/firefly-signer/Chart.yaml
Comment (EnriqueL8) [inline charts/firefly-signer/Chart.yaml:7]: which are these version not matching FF version

File: charts/firefly-signer/Chart.yaml
Comment (EnriqueL8) [inline charts/firefly-signer/Chart.yaml:11]: Update this to myself and @onelapahead please

File: charts/firefly-evmconnect/Chart.yaml
Comment (EnriqueL8) [inline charts/firefly-evmconnect/Chart.yaml:12]: update to @onelapahead please

File: charts/firefly-signer/templates/keystore.yaml
Comment (EnriqueL8) [inline charts/firefly-signer/templates/keystore.yaml:38]: Is there a script to be able to generate a key and add it? Would be nice to have, maybe a follow up issue

Comment (EnriqueL8) [reply charts/firefly-signer/templates/keystore.yaml:38]: or even a way to import keys!

File: charts/firefly-signer/templates/statefulset.yaml
Comment (EnriqueL8) [inline charts/firefly-signer/templates/statefulset.yaml:94]: Why are we not mounting the whole keystore? FYI on the above about users wanting to add new keys?

File: charts/firefly-signer/values.yaml
Comment (EnriqueL8) [inline charts/firefly-signer/values.yaml:4]: 1.3 tag? feels weird to have IfNotPresent with latest

File: charts/firefly-signer/values.yaml
Comment (EnriqueL8) [inline charts/firefly-signer/values.yaml:7]: so Besu runs in a separate namespace? why?

File: charts/firefly/README.md
Comment (EnriqueL8) [inline charts/firefly/README.md:138]: This needs updating, no longer alpha

File: charts/firefly/README.md
Comment (EnriqueL8) [inline charts/firefly/README.md:142]: This as well, not beta and it does get configured

File: charts/firefly/README.md
Comment (EnriqueL8) [inline charts/firefly/README.md:388]: Have we tested fab connect?

File: charts/firefly/values.yaml
Comment (EnriqueL8) [inline charts/firefly/values.yaml:271]: what are all these indents? some sort of linting?

File: charts/firefly/values.yaml
Comment (EnriqueL8) [inline charts/firefly/values.yaml:338]: v1.3.0?

File: charts/firefly/values.yaml
Comment (EnriqueL8) [inline charts/firefly/values.yaml:429]: v1.3.0 please, we might be in 1.3.2 even

File: charts/firefly/values.yaml
Comment (EnriqueL8) [inline charts/firefly/values.yaml:505]: ditto

File: charts/ipfs/Chart.yaml
Comment (EnriqueL8) [inline charts/ipfs/Chart.yaml:27]: remove nicko

File: hack/multiparty.sh
Comment (EnriqueL8) [inline hack/multiparty.sh:70]: I can see this failing a lot... follow up issue to raise to use the new status APIs for both the checks and progressing between states

File: values/monitoring/grafana-goquorum-dashboard.yml
Comment (EnriqueL8) [inline values/monitoring/grafana-goquorum-dashboard.yml:20]: Do we need this? Are we deploying geth?

File: values/validator.yml
Comment (EnriqueL8) [inline values/validator.yml:28]: What is this AWS and azure stuff...

**Review by EnriqueL8 — CHANGES_REQUESTED** [review]
Looking good - quite a few tweaks needed

**Comment by SamMayWork** [comment]
In discussion w/ @EnriqueL8 - there's a quite a few comments from the original PR around the app versions that we're using in these charts, this mostly appears to be leftovers from the original PR, but something that we need addressing.

**Problem: Right now we're not specifying a specific tag for the images that we're using for FireFly components, this means that we're defaulting to latest which might stop working at some point we don't know.**

The solution we're proposing is the following:
 - This PR contains the images for FireFly 1.3.0
 - When this PR is merged, we cut a release of this repository which contains only the 1.3 images
 - A GitHub action is added which does some basic testing of the stack using KiND
 - On a regular cadence we update the images that these charts are using and then cut a new release each time

Automation to make the above easier is certainly possible and would be a meaningful contribution to this project.

File: charts/firefly-evmconnect/Chart.yaml
Comment (SamMayWork) [reply charts/firefly-evmconnect/Chart.yaml:9]: Changed as per https://github.com/hyperledger/firefly-helm-charts/pull/76#issuecomment-2186008321

File: charts/firefly-signer/Chart.yaml
Comment (SamMayWork) [reply charts/firefly-signer/Chart.yaml:7]: This is the correct version that was bundled with 1.3.0 https://github.com/hyperledger/firefly-signer/pkgs/container/firefly-signer/versions?filters%5Bversion_type%5D=tagged

File: charts/firefly-evmconnect/Chart.yaml
Comment (SamMayWork) [reply charts/firefly-evmconnect/Chart.yaml]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/205f20754d1f00a68130ca2234e50bf3db86aca8

File: charts/firefly-signer/values.yaml
Comment (SamMayWork) [reply charts/firefly-signer/values.yaml]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/205f20754d1f00a68130ca2234e50bf3db86aca8

File: charts/firefly/values.yaml
Comment (SamMayWork) [reply charts/firefly/values.yaml]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/205f20754d1f00a68130ca2234e50bf3db86aca8

File: charts/firefly/values.yaml
Comment (SamMayWork) [reply charts/firefly/values.yaml]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/205f20754d1f00a68130ca2234e50bf3db86aca8

File: charts/ipfs/Chart.yaml
Comment (SamMayWork) [reply charts/ipfs/Chart.yaml]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/205f20754d1f00a68130ca2234e50bf3db86aca8

File: charts/firefly/values.yaml
Comment (SamMayWork) [reply charts/firefly/values.yaml]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/205f20754d1f00a68130ca2234e50bf3db86aca8

File: charts/firefly-signer/values.yaml
Comment (SamMayWork) [reply charts/firefly-signer/values.yaml:7]: It _used_ to but I pulled back into the default namespace with the FireFly components, this update just got missed.

Comment (SamMayWork) [reply charts/firefly-signer/values.yaml]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/d71b1fcaab0080cd86040e1c33c998b062f7aebe

**Comment by SamMayWork** [comment]
In reference to specific comments about things in the chart that aren't in the stack we're intending to use, for example AWS/Azure/Geth...

It looks like parts of the values chart that we're using come from documentation from things like Besu and purposefully have multiple values so that they're unopinionated as possible, but not sure we want that here. Essentially, these helm charts now have 2 parts: deployment of chain infrastructure, and deployment of FireFly. For chain infrastructure we want to be _very_ opinionated so that we don't break our KiND installation path, but our FireFly helm charts must be unopinionated.

File: values/validator.yml
Comment (SamMayWork) [reply values/validator.yml]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/55610c410186a352fe848b61fb5761986a4f0309

File: values/monitoring/grafana-goquorum-dashboard.yml
Comment (SamMayWork) [reply values/monitoring/grafana-goquorum-dashboard.yml]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/55610c410186a352fe848b61fb5761986a4f0309

File: charts/firefly/README.md
Comment (SamMayWork) [reply charts/firefly/README.md]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/213a84dfda8d8c097d37bd6ac7d3d2115f871048

File: charts/firefly/README.md
Comment (SamMayWork) [reply charts/firefly/README.md]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/213a84dfda8d8c097d37bd6ac7d3d2115f871048

**Comment by SamMayWork** [comment]
From comments here by @EnriqueL8 it looks like this PR being merged creates a few new items to make things here easier/more resilient:

- Automated testing of the chart through a GitHub actions (checking that components deploy properly, with maybe some basic integration testing).
- Hardening around the multiparty script using the status APIs to ensure that the nodes change through states as expected
- Script to add and manage new keys into the deployed FireFly
- Verification of the multiparty scripts against new APIs introduced into FireFly since 1.3 was released

I'll open these issues once this PR is merged

File: hack/multiparty.sh
Comment (SamMayWork) [reply hack/multiparty.sh:70]: See https://github.com/hyperledger/firefly-helm-charts/pull/76#issuecomment-2186095649

File: charts/firefly-signer/templates/statefulset.yaml
Comment (SamMayWork) [reply charts/firefly-signer/templates/statefulset.yaml:94]: See https://github.com/hyperledger/firefly-helm-charts/pull/76#issuecomment-2186095649

File: charts/firefly-signer/templates/keystore.yaml
Comment (SamMayWork) [reply charts/firefly-signer/templates/keystore.yaml:38]: See https://github.com/hyperledger/firefly-helm-charts/pull/76#issuecomment-2186095649

File: README.md
Comment (SamMayWork) [reply README.md:37]: See https://github.com/hyperledger/firefly-helm-charts/pull/76#issuecomment-2186095649

File: charts/firefly/README.md
Comment (SamMayWork) [reply charts/firefly/README.md:388]: I personally haven't, worth noting that these configuration options would be outside the scope of what the tooling could generate so we'd need to stand up a custom stack to test this out, interested to know if we see this as a blocker for this PR or if this is future work.

File: charts/firefly/values.yaml
Comment (SamMayWork) [reply charts/firefly/values.yaml:271]: Yep, it's some linting that's causing this but I can't recreate that linting on my own machine either through the `make lint` commands or by running those commands directly against the directory so I'm unsure how this was in the original PR. Both are obviously valid, but the original version is significantly more conventional so going to back out these changes.

Comment (SamMayWork) [reply charts/firefly/values.yaml]: Changed in https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/36bd1b29bf4ab472d68a82d5075b5a3f247ed74b

**Comment by SamMayWork** [comment]
OK - Calling this PR good for a re-review, it's worth noting that currently this is not going to work because of this issue https://github.com/hyperledger/firefly-dataexchange-https/issues/81, once a release has been properly cut, I'll retest this PR and we should be good for a merge.

**Comment by EnriqueL8** [comment]
I've just ran this locally

```
W0628 12:24:56.735297   28761 warnings.go:70] unknown field "spec.automountServiceAccountToken"
```

**Comment by EnriqueL8** [comment]
It would be nice if we could configure the swagger urls to be the port forward URL so that the swagger UI works with FireFly 
image


We exposed a configuration in FireFly to allow this https://hyperledger.github.io/firefly/latest/reference/config/#http

We could also think about deploying an ingress instead of port-forwarding as it's such a common thing

File: charts/firefly/Chart.yaml
Comment (EnriqueL8) [inline charts/firefly/Chart.yaml:31]: Can you remove drew please

**Comment by EnriqueL8** [comment]
Having a small tutorial on how to push to the kind registry a custom image would be amazing as well, doesn't need to be part of this PR

File: charts/firefly/Chart.yaml
Comment (SamMayWork) [reply charts/firefly/Chart.yaml]: https://github.com/hyperledger/firefly-helm-charts/pull/76/commits/ceaa246d9a659fd704cbb5a3c55f0781d7c0c7fd

**Comment by EnriqueL8** [comment]
We will hold off until 1.3.1 goes out and we will get this in!

**Review by EnriqueL8 — APPROVED** [review]
Awesome, looks good to me now that the images have been updated

**Pull request merged**
